core/storage/base_model/gae_models_test.py

```diff
@@ -235,3 +352,16 @@ def test_get_multi_versions_errors(self):
             ValueError,
             'At least one version number is invalid'):
             TestVersionedModel.get_multi_versions('model_id1', [1, 1.5, 2])
+
+    # Note: This test for the generic base model is written at last since some
```

Comment:
This is the error log:
```
ERROR: test_get_commit (core.storage.base_model.gae_models_test.CommitLogEntryModelTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/circleci/oppia/core/storage/base_model/gae_models_test.py", line 198, in test_get_commit
    self.assertEqual(test_model.version, 1)
AttributeError: 'NoneType' object has no attribute 'version'

======================================================================
ERROR: test_commit_with_model_instance_deleted_raises_error (core.storage.base_model.gae_models_test.VersionedModelTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/circleci/oppia/core/storage/base_model/gae_models_test.py", line 243, in test_commit_with_model_instance_deleted_raises_error
    model_by_version = TestVersionedModel.get_version('model_id1', 1)
  File "/home/circleci/oppia/core/storage/base_model/gae_models.py", line 697, in get_version
    cls.get(entity_id)._require_not_marked_deleted()
  File "/home/circleci/oppia/core/storage/base_model/gae_models.py", line 772, in get
    return super(VersionedModel, cls).get(entity_id, strict=strict)
  File "/home/circleci/oppia/core/storage/base_model/gae_models.py", line 102, in get
    (cls.__name__, entity_id))
EntityNotFoundError: Entity for class TestVersionedModel with id model_id1 not found

======================================================================
ERROR: test_get_multi_versions (core.storage.base_model.gae_models_test.VersionedModelTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/circleci/oppia/core/storage/base_model/gae_models_test.py", line 318, in test_get_multi_versions
    'model_id1', [1, 2])
  File "/home/circleci/oppia/core/storage/base_model/gae_models.py", line 728, in get_multi_versions
    raise ValueError('The given entity_id %s is invalid.' % (entity_id))
ValueError: The given entity_id model_id1 is invalid.

======================================================================
FAIL: test_get_multi_versions_errors (core.storage.base_model.gae_models_test.VersionedModelTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/circleci/oppia/core/storage/base_model/gae_models_test.py", line 332, in test_get_multi_versions_errors
    TestVersionedModel.get_multi_versions('model_id1', [1, 2, 3])
AssertionError: "Requested version number 3 cannot be higher than the current version number 2." does not match "The given entity_id model_id1 is invalid."

======================================================================
FAIL: test_get_snapshots_metadata_with_invalid_model_raises_error (core.storage.base_model.gae_models_test.VersionedModelTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/circleci/oppia/core/storage/base_model/gae_models_test.py", line 310, in test_get_snapshots_metadata_with_invalid_model_raises_error
    model1.get_snapshots_metadata('model_id1', [1])
AssertionError: "Invalid version number .+ for model .+ with id .+" does not match "Entity for class TestVersionedModel with id model_id1 not found"
```
I can't figure this out but this only occurs if I place the test at the beginning of the file. This also does not occur when I run tests with the exact test class, i.e --test_target=core.storage...gae_models_test.BaseModelUnitTests. It seems like the above failing tests are somehow dependent on this test.

Reply:
So I just checked out your PR and tried that, and the tests pass for me. Could you please send me the version of the file that causes the error, and the command you used to run tests, and I'll check it out? This shouldn't be happening so I would like to dig into it.

Reply:
Put this test (`test_raise_exception_by_mocking_collision`) at the end of BaseModelUnitTests. I ran bash scripts/run_backend_tests.sh --test_target=core.storage.base_model.gae_models_test

Reply:
Ah, I think you're probably running into issues because BaseModel gets subclassed in lots of places. I suggest that you make a test subclass of BaseModel, and perform the swap on that class instead.

Reply:
Done